internal/services/containers/container_registry_cache_rule_resource.go

```diff
@@ -0,0 +1,252 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package containers
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/containerregistry/2021-08-01-preview/registries"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/containerregistry/2023-07-01/cacherules"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/containers/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+)
+
+var _ sdk.Resource = ContainerRegistryCacheRule{}
+
+type ContainerRegistryCacheRule struct{}
+
+func (ContainerRegistryCacheRule) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			Description:  "The name of the cache rule.",
+			ValidateFunc: validate.ContainerRegistryCacheRuleName,
+		},
+		"container_registry_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: registries.ValidateRegistryID,
+		},
+		"credential_set_id": {
+			Type:        pluginsdk.TypeString,
+			Optional:    true,
+			Description: "The ARM resource ID of the credential store which is associated with the cache rule.",
+		},
+		"source_repo": {
+			Type:        pluginsdk.TypeString,
+			Required:    true,
+			ForceNew:    true,
+			Description: "The full source repository path such as 'docker.io/library/ubuntu'.",
+		},
+
+		"target_repo": {
+			Type:        pluginsdk.TypeString,
+			Required:    true,
+			ForceNew:    true,
+			Description: "The target repository namespace such as 'ubuntu'.",
+		},
+	}
+}
+
+func (ContainerRegistryCacheRule) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (ContainerRegistryCacheRule) ModelObject() interface{} {
+	return nil
+}
+
+func (ContainerRegistryCacheRule) ResourceType() string {
+	return "azurerm_container_registry_cache_rule"
+}
+
+func (r ContainerRegistryCacheRule) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			cacheRulesClient := metadata.Client.Containers.ContainerRegistryClient_v2023_07_01.CacheRules
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			ctx, cancel := timeouts.ForCreate(metadata.Client.StopContext, metadata.ResourceData)
+
```

Comment:
Similar to the data source we can decode the model here to access the values in the config
```suggestion

			var config ContainerRegistryCacheRuleModel
			if err := metadata.Decode(&config); err != nil {
				return err
			}

```